docs/sdk/core/baseapp.md

```diff
@@ -0,0 +1,48 @@
+# BaseApp
+
+The BaseApp defines the foundational implementation for a basic ABCI application
+so that your Cosmos-SDK application can communicate with an underlying
+Tendermint node.
+
+The BaseApp is composed of many internal components. Some of the most important
+include the `CommitMultiStore` and its internal state. The internal state is
+essentially two sub-states, both of which are used for transaction execution
+during different phases, `CheckTx` and `DeliverTx` respectively. During block
+commitment, only the `DeliverTx` is persisted.
+
+## Transaction Life Cycle
+
+During the execution of a transaction, it may pass through both `CheckTx` and
+`DeliverTx` as defined in the ABCI specification. `CheckTx` is executed by the
+proposing validator and is used for the Tendermint mempool. Notably, both
+`CheckTx` and `DeliverTx` execute the application's AnteHandler (if defined),
+where the AnteHandler is responsible for pre-message validation checks such as
+account and signature validation as well as fee deduction and collection. It is
+important to note that the AnteHandler does, and is expected to, perform various
+state transitions such as nonce incrimination and fee deduction.
+
+### CheckTx
+
+During the execution of `CheckTx`, first the AnteHandler is executed. If the
+AnteHandler fails or panics then the transaction fails. After the successful
+execution of theAnteHandler, the messages are executed. If all the messages
```

Comment:
messages are not executed, only the AnteHandler is run.

Reply:
Correct. See my above comment.